app/(landing)/views/offer/token.tsx

```diff
@@ -5,134 +5,189 @@ import Image from 'next/image'
-    </div>
+  return <div className={styles.distribution}>{children}</div>
 }
 
 export function TokenFees({ children }: PropsWithChildren) {
-  return <div className={styles.item}>
-        <span className={styles.label}>Fees</span>
-        <span className={styles.value}>{children}</span>
+  return (
+    <div className={styles.item}>
+      <span className={styles.label}>Fees</span>
+      <span className={styles.value}>{children}</span>
     </div>
+  )
 }
 
 export function TokenApy({ children }: PropsWithChildren) {
-  return <div className={styles.item}>
-        <span className={styles.label}>APY</span>
-        <span className={styles.value}>{children}</span>
+  return (
+    <div className={styles.item}>
+      <span className={styles.label}>APY</span>
+      <span className={styles.value}>{children}</span>
     </div>
+  )
 }
 
 export function TokenGrowth({ children }: PropsWithChildren) {
-  return <div className={styles.item}>
-        <span className={styles.label}>Growth</span>
-        <span className={styles.value}>{children}</span>
+  return (
+    <div className={styles.item}>
+      <span className={styles.label}>Growth</span>
+      <span className={styles.value}>{children}</span>
     </div>
+  )
 }
 
-export function Action({ development }: TokenProps) {
+export function Action({ development, balance, href, buttonLabel }: TokenProps) {
   return (
-        <div className={styles.action}>
-            <div className={styles.insurance}>
-                <span className={styles.label}>Insurance Coverage</span>
-                <span className={styles.value}>100%</span>
-            </div>
-
-            <Progress size="sm" aria-label="Insurance" value={100} classNames={{
-              indicator: styles.insuranceInducator,
-            }} />
-
-            <Button size='lg' dark className={styles.actionButton}>
-                {development ? 'Join the Waitlist' : 'Save'}
-            </Button>
-
-            {development && <p className={styles.development}>Coming soon</p>}
-        </div>
+    <div className={styles.action}>
+      {balance && (
+        <>
+          <div className={styles.balance}>
+            <span className={styles.label}>Your Deposit</span>
+            <span className={styles.value}>{balance}</span>
+          </div>
+          <div className={styles.divider} />
+        </>
+      )}
+      <div className={styles.insurance}>
+        <span className={styles.label}>Insurance Coverage</span>
+        <span className={styles.value}>100%</span>
+      </div>
+
+      <div className={styles.insuranceInducatorWrapper}>
+        <Progress
+          size="sm"
+          aria-label="Insurance"
+          value={100}
+          classNames={{
+            indicator: styles.insuranceInducator,
+          }}
+        />
+      </div>
+
+      {href ? <InternalLink href={href}>{<CTAButton />}</InternalLink> : <CTAButton />}
+
+      {development && <p className={styles.development}>Coming soon</p>}
```

Comment:
Better move this label to other place, it looks like a button, but it not a button and it not a clickable. If it label, it must looks like a label. Check design for example
![image](https://github.com/eonian-core/web/assets/11697794/f5e9b1b0-880a-4397-898d-d9a6681a25c6)

Reply:
Stop adding new elements to this card please, i disabled the button